src/wagtail_personalisation/adapters.py

```diff
@@ -177,30 +177,25 @@ def refresh(self):
         still apply to the requesting visitor.
 
         """
-        enabled_segments = Segment.objects.filter(status=Segment.STATUS_ENABLED)
-        persistent_segments = enabled_segments.filter(persistent=True)
+        all_segments = Segment.objects.filter(status=Segment.STATUS_ENABLED)
 
-        session_segments = self.get_segments()
+        current_segments = self.get_segments()
         rules = AbstractBaseRule.__subclasses__()
 
-        # Create a list to store the new request session segments and
-        # re-apply all persistent segments (if they are still enabled).
-        new_segments = [segment for segment in session_segments
-                        if persistent_segments.filter(id=segment['id']).exists()]
-
         # Run tests on all remaining enabled segments to verify applicability.
-        for segment in enabled_segments:
+        additional_segments = []
+        for segment in all_segments:
             segment_rules = []
             for rule in rules:
                 segment_rules += rule.objects.filter(segment=segment)
-            result = self._test_rules(segment_rules, self.request,
-                                      match_any=segment.match_any)
+
+            result = self._test_rules(
+                segment_rules, self.request, match_any=segment.match_any)
 
             if result:
-                segdict = create_segment_dictionary(segment)
-                if not any(seg['id'] == segdict['id'] for seg in new_segments):
-                    new_segments.append(segdict)
+                additional_segments.append(segment)
 
+        new_segments = current_segments + additional_segments
         self.set_segments(new_segments)
```

Comment:
Defining `new_segments` seems redundant, you can just write `self.set_segments(current_segments + additional_segments)`